### PR TITLE
Make connecting to DB at KB init optional

### PR DESCRIPTION
**Pull request by rmitsch**
## Description

`WikiKB` so far assumed that a DB was available at initialization time. This might no be always the case, so this PR adds a feature flag to make connecting to the DB at initialization time optional and exposes a method to do this at any time.

Split off of https://github.com/explosion/wikid/pull/6.

### Types of change

Enhancement.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: src/kb.py [hunk 1]
```diff
@@ -63,23 +65,32 @@ def __init__(
...
+            WikiKB.establish_db_connection() has to be called explicitly before any other operations are invoked.
         """
         super().__init__(vocab, entity_vector_length)
 
         self._paths = {"db": db_path, "annoy": annoy_path}
         self._language = language
         self._annoy: Optional[annoy.AnnoyIndex] = None
         self._n_trees = n_trees
-        self._db_conn = establish_db_connection(language, self._paths["db"])
+        self._db_conn: Optional[sqlite3.Connection] = None
```
Comment (svlandeg) [inline src/kb.py:78]: Let's add an additional helper method, something like `ensure_connection`, that raises a custom error when `self._db_conn` is called in a method but it's still `None`. Then we can call `ensure_connection` up front at the start of any such method.

Comment (rmitsch) [reply src/kb.py:78]: Done in https://github.com/explosion/wikid/pull/18/commits/318615aa1ac832feb9abd1637fbf8c0060102ede.

File: test_wikikb.py [hunk 2]
```diff
@@ -218,3 +218,11 @@ def test_kb_serialization(_kb, method: str) -> None:
             )
         ]
     )
+
+
+def test_kb_wo_db(_kb) -> None:
```
Comment (svlandeg) [inline test_wikikb.py]:
```suggestion
def test_kb_without_db(_kb) -> None:
```

**Review by svlandeg — APPROVED** [review]
Feel free to merge once the CI goes green!

**Pull request merged**